http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProvider.scala

```diff
@@ -127,6 +127,63 @@ class SamAuthProvider[F[_]: OpenTelemetryMetrics](
     } yield (callerActions, projectCallerActions)
   }
 
+  /**
+   * Get all Sam resource IDs of the given type on which the user is granted supported permissions. Returns explicitly
+   * granted policies only; users with workspace owner access, for example, can see all runtimes in the workspace, but
+   * getAuthorizedIds[SamRuntimeResourceId] is not guaranteed to return all runtimes in that workspace. Works for any
+   * Sam resource type except SharedApp and similar alternative app types; these require a specific Sam resource object
+   * to determine permissions, as these are stored on the individual records.
+   * @param isOwner whether to return only resources that the user is granted ownership on (has `ownerRoleName`,
+   * and `ownerRoleName` is an element in `policyNames` on the SamResource)
+   * @param userInfo the current user
+   * @param samResource the resource archetype (App, Runtime, etc)
+   * @param decoder converts Json -> SamResourceId, needed by SamDao
+   * @param ev the request context
+   * @tparam R SamResourceId
+   * @return List[SamResourceId] of all resources of type R which are visible to the user at the given access level
+   */
+  override def getAuthorizedIds[R <: SamResourceId](
+    isOwner: Boolean,
+    userInfo: UserInfo
+  )(implicit
+    samResource: SamResource[R],
+    decoder: Decoder[R],
+    ev: Ask[F, TraceId]
+  ): F[Set[R]] = {
+    val authHeader = Authorization(Credentials.Token(AuthScheme.Bearer, userInfo.accessToken.token))
+    val ownerRole: SamRole = samResource.ownerRoleName
+    val canOwnerRead: Boolean = samResource.policyNames.isEmpty || samResource.policyNames.exists {
+      policyName: SamPolicyName =>
+        ownerRole.asString == policyName.toString
```

Comment:
Based on the [comments here in slack about what we are modeling here](https://sam.dsde-dev.broadinstitute.org/#/Resources/listResourcePoliciesV2) - 
I feel like and am not sure if we are comparing apples and oranges. 

Hoping to have @rtitle and/or @Qi77Qi review this for clarity - the team would really like to get this out to re-enable the checkbox and address tech-debt in subsequent units of work.

Reply:
It is a bit apples and oranges… the owner role is defined in Sam conf (looks like it's [creator](https://github.com/broadinstitute/sam/blob/develop/src/main/resources/reference.conf#L756) for notebook-cluster resources). And the policy name is whatever we pass to Sam when creating the resource. I think though in practice we use [createResourceWithDefaultsV2](https://sam.dsde-dev.broadinstitute.org/#/Resources/createResourceWithDefaultsV2) which happens to create a resource policy with the same name as the owner role.

So it probably works but the model is a bit confusing. Also I still need to grok the intent of this code a little more -- I do wonder if the `isOwner` / `canOwnerRead` decisioning should be implicit in the Sam query. I see we're meeting next week -- I'll do a little homework to come prepared with a better opinion.

Reply:
This was unclear configuration logic; replaced with simplified logic.